### ISSUE TITLE
Performance profiling

### ISSUE THREAD

**Issue by fire** [godot-extended-libraries/hydro#5]
Are you able to create a stress scene so I can run a profile?

**Comment by jonri** [comment]
Yes, certainly!  To start with I'm thinking we should have a scene with one high-poly object and a scene with many lower-poly objects.  In addition to getting a baseline profile, those are the two most obvious areas that we could consider running the calculations in parallel and we'd be able to use these scenes to evaluate any improvement from doing so.

The other potential bottleneck to test would be the wave height gdscript callback function, so we'll also want a scene with an object under no wave function, a constant wave function, and a complex one.

**Comment by jonri** [comment]
@fire Let me know if you think I'm going down the right path with this:

[StressTest.zip](https://github.com/godot-extended-libraries/hydro/files/5432136/StressTest.zip)

- I have scenes with 1 object and 10 objects, in both cases the total number of faces is about the same so we can compare the results with each other.
- I have scenes with a built-in wave height, and two with simple and complex callbacks to gdscript for the wave height.
- I took a quick look with the built-in profiler and these all take about 125% of a physics timeslot each cycle on my Ryzen 3600.  I can bump it up easily if needed.


So far, all the scenes test with an object totally underwater, this isolates and makes consistent the code to calculate the forces.  We'll want one with objects on the surface too, in order to hit the mesh clipping function too.  I was also going to add a scene with 100 objects but the same face count.

Once we have a nice suite, these might be able to double as both performance and regression tests.  If these are in line with what you're looking for, I think they'll warrant inclusion in the repo so that anyone could perform the same tests.

**Comment by jonri** [comment]
@fire I decided to convert this into a script that allows different numbers of objects and faces to be specified.  When the objects sink to a specified depth, the script will print some stats and exit.  This should allow for more versatility in running this from various profilers, automated test suites, etc.  I've added a small section to the [wiki](https://github.com/godot-extended-libraries/hydro/wiki#testing) on how to use it.

Let me know if https://github.com/godot-extended-libraries/hydro/commit/d9f4aa19a28dbd91d195b2f36a49a8499d053f76 is sufficient for your profiling work, I used it with `perf` on Linux and have already found some interesting results.
